src/pytest_cov/engine.py

```diff
@@ -18,7 +18,7 @@ def __init__(self, cov_source, cov_report, cov_config, cov_append, config=None,
         """Get some common config used by multiple derived classes."""
         self.cov_source = cov_source
         self.cov_report = cov_report
-        self.cov_config = os.path.abspath(cov_config)
+        self.cov_config = os.path.abspath(cov_config) if os.path.exists(cov_config) else cov_config
```

Comment:
If you're not specifying `--cov-config` then `cov_config` will be `None` (and coverage will try to find it itself).

If you do specify it and it doesn't exist then coverage should complain, I suspect it doesn't and we should do our own check.

You're quite correct here, if coverage is using a autofound config file with a different name this will make the coverage reinit malfunction (since config won't exist in cwd).